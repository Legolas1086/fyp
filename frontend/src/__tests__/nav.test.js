import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Nav from '../nav'
const mockedNavigate = jest.fn("1");
jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useLocation: ()=>mockedNavigate,
}))
describe('snapshot testing',()=>{
    test("renders approve the request compnenet",()=>{
        const fragment = render(
            <BrowserRouter>
            <Nav/>
            </BrowserRouter>
        ).asFragment();
    expect(fragment).toMatchSnapshot()
    })
});