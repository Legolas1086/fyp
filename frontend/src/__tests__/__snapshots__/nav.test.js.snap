// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing renders approve the request compnenet 1`] = `
<DocumentFragment>
  <nav
    class="Nav_bar"
  >
    <h1>
      Pustak Bandaar
    </h1>
    <div
      class="nav-links"
      tabindex="1"
    >
      <span
        style="padding: 1rem; border-radius: 10px;"
      >
        <a
          class="nav-button"
          href="/"
        >
          Dashboard
        </a>
      </span>
      <span
        style="padding: 1rem; border-radius: 10px;"
      >
        <a
          class="nav-button"
          href="/post"
        >
          Post 
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-square-plus "
            data-icon="square-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 32C419.3 32 448 60.65 448 96V416C448 451.3 419.3 480 384 480H64C28.65 480 0 451.3 0 416V96C0 60.65 28.65 32 64 32H384zM224 368C237.3 368 248 357.3 248 344V280H312C325.3 280 336 269.3 336 256C336 242.7 325.3 232 312 232H248V168C248 154.7 237.3 144 224 144C210.7 144 200 154.7 200 168V232H136C122.7 232 112 242.7 112 256C112 269.3 122.7 280 136 280H200V344C200 357.3 210.7 368 224 368z"
              fill="currentColor"
            />
          </svg>
        </a>
      </span>
      <span
        style="padding: 1rem; border-radius: 10px;"
      >
        <a
          class="nav-button"
          href="/mybooks"
        >
          MyBooks
        </a>
      </span>
      <span
        style="padding: 1rem; border-radius: 10px;"
      >
        <a
          class="nav-button"
          href="/chat"
        >
          Messages
        </a>
      </span>
      <span
        style="padding: 1rem; border-radius: 10px;"
      >
        <a
          class="nav-button"
          href="/login"
        >
          Logout
        </a>
      </span>
    </div>
  </nav>
</DocumentFragment>
`;
